### PR TITLE
981 BigDecimal equality verification util

### PR DESCRIPTION
**Pull request by jsanhc**
- Updated equals method for all OB objects where BigDecimal is used
- Added new EqualityUtil
- Updated pom descriptor junit 5 dependency to import all juniper dependencies to build the unit tests
- Created unit test for EqualityUtil
- Created unit test where is checked all OB objects updated
    
Issue: https://github.com/SecureApiGateway/SecureApiGateway/issues/981

### CODE REVIEW

File: secure-api-gateway-ob-uk-common-obie-datamodel/src/main/java/uk/org/openbanking/datamodel/account/OBCurrencyExchange5.java [hunk 1]
```diff
@@ -210,7 +212,8 @@ public boolean equals(Object o) {
         return Objects.equals(this.sourceCurrency, obCurrencyExchange5.sourceCurrency) &&
                 Objects.equals(this.targetCurrency, obCurrencyExchange5.targetCurrency) &&
                 Objects.equals(this.unitCurrency, obCurrencyExchange5.unitCurrency) &&
-                Objects.equals(this.exchangeRate, obCurrencyExchange5.exchangeRate) &&
+                // TODO: temporary fix for https://github.com/SecureApiGateway/SecureApiGateway/issues/981
+                isEqual(this.exchangeRate, obCurrencyExchange5.exchangeRate) &&
```
Comment (jorgesanchezperez) [inline secure-api-gateway-ob-uk-common-obie-datamodel/src/main/java/uk/org/openbanking/datamodel/account/OBCurrencyExchange5.java:216]: Modification in `equals` method to verify the equality of BigDecimals.

`isEqual(BigDecimal compareTo, BigDecimal toCompare)` imported as static, used in all OB objects that have a BigDecimal field.

File: secure-api-gateway-ob-uk-common-obie-datamodel/src/test/java/uk/org/openbanking/datamodel/utils/EqualityBigDecimalUtilTest.java [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+                        new OBLoanInterestTierBand1().loanProviderInterestRate(new BigDecimal("56.0120"))
+                                .tierValueMaximum(new BigDecimal("2.20"))
+                                .tierValueMaxTerm(new BigDecimal("5.2"))
+                                .tierValueMinimum(new BigDecimal("4.1300"))
+                                .tierValueMinTerm(new BigDecimal("7.30")),
+                        new OBLoanInterestTierBand1().loanProviderInterestRate(new BigDecimal("56.012000"))
+                                .tierValueMaximum(new BigDecimal("2.2"))
+                                .tierValueMaxTerm(new BigDecimal("5.200"))
+                                .tierValueMinimum(new BigDecimal("4.13"))
+                                .tierValueMinTerm(new BigDecimal("7.3000"))
```
Comment (dbadham-fr) [inline secure-api-gateway-ob-uk-common-obie-datamodel/src/test/java/uk/org/openbanking/datamodel/utils/EqualityBigDecimalUtilTest.java:59]: Nice, very thorough testing!

File: secure-api-gateway-ob-uk-common-obie-datamodel/src/test/java/uk/org/openbanking/datamodel/utils/EqualityVerificationUtilTest.java [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static uk.org.openbanking.datamodel.utils.EqualityVerificationUtil.*;
+
+/**
+ * Unit test for {@link EqualityVerificationUtil}
+ */
+public class EqualityVerificationUtilTest {
+
+    private static Stream<Arguments> argumentsProviderTrue() {
```
Comment (dbadham-fr) [inline secure-api-gateway-ob-uk-common-obie-datamodel/src/test/java/uk/org/openbanking/datamodel/utils/EqualityVerificationUtilTest.java:33]: can you add a test case checking testing scientific notation please

**Review by dbadham-fr — CHANGES_REQUESTED** [review]
Very nice work.

One minor comment, and also can you add a little note in the README code gen section about this change. Something along the lines of: "You will need to edit .equals methods of data model objects containing big decimals, see PR (and a link to this PR)"

File: secure-api-gateway-ob-uk-common-obie-datamodel/src/test/java/uk/org/openbanking/datamodel/utils/EqualityVerificationUtilTest.java (hunk 3 above)
Comment (jorgesanchezperez) [reply secure-api-gateway-ob-uk-common-obie-datamodel/src/test/java/uk/org/openbanking/datamodel/utils/EqualityVerificationUtilTest.java:35]: Done

File: secure-api-gateway-ob-uk-common-obie-datamodel/src/test/java/uk/org/openbanking/datamodel/utils/EqualityBigDecimalUtilTest.java (hunk 2 above)
Comment (jorgesanchezperez) [reply secure-api-gateway-ob-uk-common-obie-datamodel/src/test/java/uk/org/openbanking/datamodel/utils/EqualityBigDecimalUtilTest.java:59]: Added new scientist annotation test.

**Pull request merged**